interactive_engine/executor/ir/graph_proxy/src/apis/graph/element/path.rs

```diff
@@ -26,46 +25,81 @@ use ir_common::generated::algebra::path_expand::ResultOpt;
 use ir_common::generated::results as result_pb;
 use ir_common::{LabelId, NameOrId};
 use pegasus::codec::{Decode, Encode, ReadExt, WriteExt};
-use pegasus_common::downcast::*;
+use pegasus_common::downcast::Any;
+use pegasus_common::downcast::AsAny;
 use pegasus_common::impl_as_any;
 
-use crate::apis::{Element, GraphElement, PropertyValue, Vertex, ID};
+use crate::apis::{Edge, Element, GraphElement, PropertyValue, Vertex, ID};
+
+#[derive(Clone, Debug, Hash, PartialEq, PartialOrd)]
+pub enum VertexOrEdge {
+    V(Vertex),
+    E(Edge),
+}
+
+impl From<Vertex> for VertexOrEdge {
+    fn from(v: Vertex) -> Self {
+        Self::V(v)
+    }
+}
+
+impl From<Edge> for VertexOrEdge {
+    fn from(e: Edge) -> Self {
+        Self::E(e)
+    }
+}
+
+impl VertexOrEdge {
+    pub fn as_vertex(&self) -> Option<&Vertex> {
+        match self {
+            VertexOrEdge::V(v) => Some(v),
+            _ => None,
+        }
+    }
+
+    pub fn as_edge(&self) -> Option<&Edge> {
+        match self {
+            VertexOrEdge::E(e) => Some(e),
+            _ => None,
+        }
+    }
+}
 
 #[derive(Clone, Debug)]
 pub enum GraphPath {
-    AllV(Vec<Vertex>),
-    SimpleAllV(Vec<Vertex>),
-    EndV((Vertex, usize)),
-    SimpleEndV((Vertex, Vec<ID>)),
+    AllPath(Vec<VertexOrEdge>),
```

Comment:
Then you may want to comment these options

Reply:
Now that with Copilot, it is very easy to add comments to the code. So comment whenever you can, on both `struct`, `function` and `enum options`